lib/storage/src/content_manager/alias_mapping.rs

```diff
@@ -0,0 +1,100 @@
+use crate::content_manager::errors::StorageError;
+use segment::common::file_operations::{atomic_save_json, read_json};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::fs;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+
+pub const ALIAS_MAPPING_CONFIG_FILE: &str = "data.json";
+
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
+struct AliasMapping(HashMap<String, String>);
+
+impl AliasMapping {
+    // TODO error type conversion
+    pub fn load(path: &Path) -> Result<Self, StorageError> {
+        read_json(path).map_err(|err| {
+            println!("failed to read_json {:?}", err);
+            StorageError::ServiceError {
+                description: "something bad happened".to_string(),
+            }
+        })
+    }
+
+    // TODO error type conversion
+    pub fn save(&self, path: &Path) -> Result<(), StorageError> {
+        atomic_save_json(path, self).map_err(|err| {
+            println!("failed to save_json {:?}", err);
+            StorageError::ServiceError {
+                description: "something bad happened".to_string(),
+            }
+        })
+    }
+}
+
+/// Not thread-safe, accesses must be synchronized by an exclusive lock at the call site
+pub struct AliasPersistence {
+    data_path: PathBuf,
+}
+
+impl AliasPersistence {
+    pub fn get_config_path(path: &Path) -> PathBuf {
+        path.join(ALIAS_MAPPING_CONFIG_FILE)
+    }
+
+    fn init_file(dir_path: &Path) -> Result<PathBuf, StorageError> {
+        let data_path = Self::get_config_path(dir_path);
+        if !data_path.exists() {
+            let mut file = fs::File::create(&data_path)?;
+            let empty_json = "{}";
+            file.write_all(empty_json.as_bytes())?;
+        }
+        Ok(data_path)
+    }
+
+    pub fn open(dir_path: PathBuf) -> Result<Self, StorageError> {
+        if !dir_path.exists() {
+            fs::create_dir_all(&dir_path)?;
+        }
+        let data_path = Self::init_file(&dir_path)?;
+        Ok(AliasPersistence { data_path })
+    }
+
+    pub fn get(&self, key: &str) -> Result<Option<String>, StorageError> {
+        let alias = AliasMapping::load(&self.data_path)?;
```

Comment:
There are not much data expected in the alias storage, so I guess we can avoid loading mapping from the disc each time we make a query. Let's just keep whole mapping in memory and persist changes on write

Reply:
The mapping is now kept in memory and persisted on write https://github.com/qdrant/qdrant/pull/402/commits/64ad8ae4445cdf2d431f3ddffbc5d1c7e254e486